plugins/amdgpu/amdgpu_plugin.c

```diff
@@ -390,53 +399,75 @@ int amdgpu_plugin_dump_file(int fd, int id)
 		    KFD_IOC_ALLOC_MEM_FLAGS_VRAM ||
 		    (bo_bucket_ptr)[i].bo_alloc_flags &
 		    KFD_IOC_ALLOC_MEM_FLAGS_GTT) {
+			if ((e->bo_info_test[i])->bo_alloc_flags &
+			    KFD_IOC_ALLOC_MEM_FLAGS_PUBLIC) {
+				pr_info("amdgpu_plugin: large bar read possible\n");
+				addr = mmap(NULL,
+					    (bo_bucket_ptr)[i].bo_size,
+					    PROT_READ,
+					    MAP_SHARED,
+					    drm_fd,	/* mapping on local gpu for prototype */
+					    (bo_bucket_ptr)[i].bo_offset);
+				if (addr == MAP_FAILED) {
+					pr_perror("amdgpu_plugin: mmap failed");
+					fd = -EBADFD;
+					close(drm_fd);
+					goto failed;
+				}
 
-			pr_info("Now try reading BO contents with /proc/pid/mem\n");
-			if (asprintf (&fname, PROCPIDMEM, helper_args.task_pid) < 0) {
-				pr_perror("failed in asprintf, %s", fname);
-				ret = -1;
-				goto failed;
-			}
+				/* direct memcpy is possible on large bars */
+				memcpy((e->bo_info_test[i])->bo_rawdata.data,
+				       addr, bo_bucket_ptr[i].bo_size);
+				munmap(addr, bo_bucket_ptr[i].bo_size);
 
-			mem_fd = open (fname, O_RDONLY);
-			if (mem_fd < 0) {
-				pr_perror("Can't open %s for pid %d", fname, helper_args.task_pid);
-				free (fname);
-				ret = -1;
-				goto failed;
-			}
+			} else {
+				pr_info("Now try reading BO contents with /proc/pid/mem\n");
+				if (asprintf (&fname, PROCPIDMEM, e->pid) < 0) {
+					pr_perror("failed in asprintf, %s", fname);
+					ret = -1;
+					goto failed;
+				}
 
-			pr_info("Opened %s file for pid = %d\n", fname, helper_args.task_pid);
-			free (fname);
+				mem_fd = open (fname, O_RDONLY);
+				if (mem_fd < 0) {
+					pr_perror("Can't open %s for pid %d", fname, e->pid);
+					free (fname);
+					ret = -1;
+					goto failed;
+				}
 
-			if (lseek (mem_fd, (off_t) (bo_bucket_ptr)[i].bo_addr, SEEK_SET) == -1) {
-				pr_perror("Can't lseek for bo_offset for pid = %d", helper_args.task_pid);
-				ret = -1;
-				goto failed;
-			}
-			pr_info("Try to read file now\n");
-
-			if (read(mem_fd, local_buf,
-				 (e->bo_info_test[i])->bo_size) !=
-			    (e->bo_info_test[i])->bo_size) {
-				pr_perror("Can't read buffer");
-				ret = -1;
-				goto failed;
-			}
-			pr_info("log initial few bytes of the raw data for this BO\n");
+				pr_info("Opened %s file for pid = %d\n", fname, e->pid);
+				free (fname);
+				if (lseek (mem_fd, (off_t) (bo_bucket_ptr)[i].bo_addr, SEEK_SET) == -1) {
+					pr_perror("Can't lseek for bo_offset for pid = %d", e->pid);
+					ret = -1;
+					goto failed;
+				}
+				pr_info("Try to read file now\n");
 
-			for (int i = 0; i < 10; i ++)
-			{
-				pr_info("0x%llx\n",((__u64*)local_buf)[i]);
-			}
-			close(mem_fd);
-			memcpy((e->bo_info_test[i])->bo_rawdata.data,
-			       (uint8_t*)local_buf,
-			       (e->bo_info_test[i])->bo_size);
-			xfree(local_buf);
+				if (read(mem_fd, local_buf,
+					 (e->bo_info_test[i])->bo_size) !=
+				    (e->bo_info_test[i])->bo_size) {
```

Comment:
```C
bo_size = read(mem_fd, local_buf, (e->bo_info_test[i])->bo_size);
if (bo_size != (e->bo_info_test[i])->bo_size) {
...
```

Reply:
We are restructuring this file as our kernel mode APIs are being reviewed and getting updated, we will take case of this and some extra prints like you mentioned in other comments below soon.